docs/architecture/adr-069-flexible-node-intitalization.md

```diff
@@ -0,0 +1,115 @@
+# ADR 069: Flexible Node Initialization
+
+## Context
+
+In an effort to support [Go-API-Stability](./adr-060-go-api-stability.Md),
+during the 0.35 development cycle, we have attempted to reduce the total size
+of the API surface area by moving most of the interface of the `node` package
+into unexported functions, as well as moving the reactors to an `internal`
+package. Having this coincide with the 0.35 release made a lot of sense
+because these interfaces were _already_ changing as a result of the `p2p`
+[refactor](./adr-061-p2p-refactor-scope.md), so it made sense to think a bit
+more about how tendermint exposes this API.
+
+It is not immediately intuitive that the nexus of this change is in the
+relationship between the `rpc` package and the `node` package, particularly
+with regards to how the node instances are started. However, this is indeed
+the case and the existing dependencies are tangled around supporting these
+cases.
+
+## Observations
+
+To prepare for this work I've been playing in [this
+branch](https://github.com/tendermint/tendermint/tree/tychoish/scratch-node-minimize),
+with the node implementation to see if, without making major
+modifications or changing interfaces, what aspects could be
+simplified. My conclusions, generally: 
+
+- the implementation of the `node` service, doesn't really need the
+  ability to introspect the implementations of reactors in most
+  cases.
+
+- the exception to this, is the cluster of
+  consensus/statesync/blockhain (which depend on eachother,) because
+  `node` itself contains some of their specific initialization logic.
```

Comment:
I'm not sure I understand why needing to make that distinction is necessary. All these reactors are services as they fulfill the `Service` interface, and they're also all processes in that they do "something".

Reply:
I think bez is correct, though it's true that in order for the "catchup feature" (pause consensus, run fastsync, switch to consensus) those services need to "know" about each other, in a way that makes the fastsync parts effectively part of consensus (from the perspective of the node.) 

This isn't hard/bad, it just means the service-based view of the node doesn't mirror the code organization as well as it does for the other reactors.

Reply:
Because making these distinctions makes it easier to reason about starting a tendermint node. You run the necessary processes (and by processes I mean a task that has a predefined termination clause) until it lets the node that it is finished and that we have reached the desired state and then you run all services. This way reactors don't need to worry when one is finished or not; the node takes care of orchestrating all that.

Reply:
Right, but those processes are run by services...

Reply:
Yeah they could be - depending on how we wanted to design it. Unlike other reactors like mempool, evidence and consensus where you're just gossiping data, blockchain and statesync have more one way data flows where a peer is either requesting data from you or providing data for you hence we could define a process as something that constantly requests data until it's finished and a service as a something that continuously provides data for peers (just thinking aloud)

Reply:
@cmwaters I think this mostly makes sense, and I don't think that it's incompatible  with any of the work discussed here, but it does feel more expansive/larger in scope than what I think we need to accomplish in the next few weeks in support of 0.35.